test/jdk/javax/swing/border/LineBorder/ScaledLineBorderTest.java

```diff
@@ -260,28 +273,33 @@ private static void showFrame(JComponent content) {
         frame.setVisible(true);
     }
 
-    private static void paintToImages(final JComponent content,
-                                      final boolean saveImages) {
-        for (double scaling : scales) {
+    private static List<TestImage> paintToImages(final TestUI testUI,
+                                                 final boolean saveImages) {
+        final List<TestImage> images = new ArrayList<>(scales.length);
+        final JComponent content = testUI.content;
+        for (double scale : scales) {
             BufferedImage image =
-                    new BufferedImage((int) Math.ceil(content.getWidth() * scaling),
-                            (int) Math.ceil(content.getHeight() * scaling),
+                    new BufferedImage((int) Math.ceil(content.getWidth() * scale),
+                            (int) Math.ceil(content.getHeight() * scale),
                             BufferedImage.TYPE_INT_ARGB);
 
             Graphics2D g2d = image.createGraphics();
-            g2d.scale(scaling, scaling);
+            g2d.scale(scale, scale);
             content.paint(g2d);
             g2d.dispose();
 
             if (saveImages) {
-                saveImage(image, getImageFileName(scaling));
+                saveImage(image, getImageFileName(scale, testUI.thickness));
             }
-            images.add(image);
+            images.add(new TestImage(image, testUI.panelLocations,
+                    scale, testUI.thickness));
         }
+        return images;
     }
 
-    private static String getImageFileName(final double scaling) {
-        return String.format("test%.2f.png", scaling);
+    private static String getImageFileName(final double scaling,
+                                           int thickness) {
+        return String.format("test%02d@%.2f.png", thickness, scaling);
```

Comment:
```suggestion
    private static String getImageFileName(final double scaling,
                                           final int thickness) {
        return String.format("test%02d@%.2f.png", thickness, scaling);
```
Declare both parameters as `final` to be consistent.